instrumentation/github.com/aws/aws-lambda-go/otellambda/README.md

```diff
@@ -0,0 +1,117 @@
+# OpenTelemetry AWS Lambda Instrumentation for Golang
+
+[![Go Reference][goref-image]][goref-url]
+[![Apache License][license-image]][license-url]
+
+This module provides instrumentation for [`AWS Lambda`](https://docs.aws.amazon.com/lambda/latest/dg/golang-handler.html).
+
+## Installation
+
+```bash
+go get -u go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda
+```
+
+## example
+See [./example](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/github.com/aws/aws-lambda-go/otellambda/example)
+
+## Usage
+
+Create a sample Lambda Go application such as below.
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+type MyEvent struct {
+	Name string `json:"name"`
+}
+
+func HandleRequest(ctx context.Context, name MyEvent) (string, error) {
+	return fmt.Sprintf("Hello %s!", name.Name ), nil
+}
+
+func main() {
+	lambda.Start(HandleRequest)
+}
+```
+
+Now use the provided wrapper to instrument your basic Lambda function:
+```go
+// Add import
+import "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda"
+
+// wrap lambda handler function
+func main() {
+	lambda.Start(otellambda.WrapHandlerFunction(HandleRequest))
```

Comment:
Not sure about `NewHandler` since we're not instantiating a new function handler, we are indeed just wrapping it. I think `InstrumentHandler()` makes sense but would like to have @bhautikpip weigh in.